internal/services/recommendation_poller.go

```diff
@@ -0,0 +1,126 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/go-playground/validator/v10"
+
+	"github.com/redhatinsights/ros-ocp-backend/internal/logging"
+	"github.com/redhatinsights/ros-ocp-backend/internal/model"
+	"github.com/redhatinsights/ros-ocp-backend/internal/types"
+	"github.com/redhatinsights/ros-ocp-backend/internal/utils"
+	"github.com/redhatinsights/ros-ocp-backend/internal/utils/kruize"
+)
+
+func FetchRecommendations(msg *kafka.Message, consumer_object *kafka.Consumer) {
+	log := logging.GetLogger()
+	validate := validator.New()
+	var kafkaMsg types.RecommendationKafkaMsg
+
+	if !json.Valid([]byte(msg.Value)) {
+		log.Errorf("received message on kafka topic is not vaild JSON: %s", msg.Value)
+		return
+	}
+	if err := json.Unmarshal(msg.Value, &kafkaMsg); err != nil {
+		log.Errorf("unable to decode kafka message: %s", msg.Value)
+		return
+	}
+	if err := validate.Struct(kafkaMsg); err != nil {
+		log.Errorf("invalid kafka message: %s", err)
+		return
+	}
+	log = logging.Set_request_details_recommendations(kafkaMsg)
+
+	var poll_cycle_complete bool = false
+	var poll_for_recommendation bool = false
+
+	var experiment_name string = kafkaMsg.Metadata.Experiment_name
+	var maxEndTimeFromReport time.Time = kafkaMsg.Metadata.Max_endtime_report
+
+	var workloadID uint = kafkaMsg.Metadata.Workload_id
+	var orgID string = kafkaMsg.Metadata.Org_id
+	var isNewRecord bool = kafkaMsg.Metadata.New_record
+
+	if isNewRecord {
+		poll_for_recommendation = true
+	} else {
+		recommendation_stored_in_db, err := model.GetFirstRecommendationSetsByWorkloadID(workloadID)
+		if err != nil {
+			log.Errorf("Error while checking for recommendation_set record: %s", err)
+		}
+		duration := maxEndTimeFromReport.Sub(recommendation_stored_in_db.MonitoringEndTime.UTC())
```

Comment:
Do we again need to depend on duration here? As we already have filtered it from report_processor?

Reply:
We needed to, for graceful msg handling, but current changes should resolve this query better